pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -55,5 +55,6 @@ def on_trainer_init(self, profiler: Union[BaseProfiler, str]):
         self.trainer.profiler = profiler or PassThroughProfiler()
 
     def on_train_start(self, trainer):
-        local_rank = trainer.local_rank if trainer.world_size > 1 else None
-        self.trainer.profiler.on_train_start(local_rank)
+        if not isinstance(trainer.profiler, PassThroughProfiler):
+            local_rank = trainer.local_rank if trainer.world_size > 1 else None
+            self.trainer.profiler.on_train_start(local_rank=local_rank, log_dir=self.trainer.log_dir)
```

Comment:
why not implement this in the `PassThroughProfiler` and make it a no-op so we don't need to special-case it here?

also why not make the local rank always an int? we could set it as 0 instead of `None`

Reply:
It is being provided as None, so the profiler can distinguish between multi processes and 1 process and adapt its internal logic for it.